### PR TITLE
ShoppingCart

### PR DESCRIPTION
**Pull request by VolandevlodD**
- Added endpoints for shopping cart management:
  - GET `/api/cart` to retrieve user's shopping cart.
  - POST `/api/cart` to add a book to the shopping cart.
  - PUT `/api/cart/cart-items/{cartItemId}` to update the quantity of a book in the shopping cart.
  - DELETE `/api/cart/cart-items/{cartItemId}` to remove a book from the shopping cart.
- Created entities for shopping cart management:
  - ShoppingCart: Represents a user's shopping cart with fields: id, user, cartItems.
  - CartItem: Represents an item in a user's shopping cart with fields: id, shoppingCart, book, quantity.
- Implemented security requirements for users with role USER.
- Utilized Liquibase for database management.
- Integrated Pagination, Sorting, and Swagger to all controllers.

### CODE REVIEW

File: src/main/java/org/example/bookstore/dto/cartitem/UpdateCartItemRequestDto.java [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package org.example.bookstore.dto.cartitem;
+
+import jakarta.validation.constraints.Min;
+import jakarta.validation.constraints.NotNull;
+
+public record UpdateCartItemRequestDto(
+        @NotNull
+        @Min(1)
```
Comment (azzilian) [inline src/main/java/org/example/bookstore/dto/cartitem/UpdateCartItemRequestDto.java]: I would add message for those annotations

**Review by azzilian — CHANGES_REQUESTED** [review]
Splendid! Trivial thing to fix

File: src/main/java/org/example/bookstore/controller/ShoppingCartController.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+                                 @RequestBody CreateCartItemRequestDto requestDto) {
+        Long userId = getUserIdFromAuthentication(authentication);
+        return cartItemService.save(requestDto, userId);
+    }
+
+    @Operation(summary = "Update a cart item", description = "Update an existing item in the "
+            + "shopping cart for the authenticated user")
+    @PutMapping("/cart-items/{cartItemId}")
+    @PreAuthorize("hasRole('USER')")
+    public CartItemDto updateCart(@RequestBody UpdateCartItemRequestDto requestDto,
```
Comment (karma-o) [inline src/main/java/org/example/bookstore/controller/ShoppingCartController.java]: same for other endpoints and their bodies
```suggestion
    public CartItemDto updateCart(@RequestBody @Valid UpdateCartItemRequestDto requestDto,
```

File: src/main/java/org/example/bookstore/model/CartItem.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+import jakarta.persistence.Id;
+import jakarta.persistence.JoinColumn;
+import jakarta.persistence.ManyToOne;
+import jakarta.persistence.Table;
+import lombok.Getter;
+import lombok.Setter;
+
+@Entity
+@Getter
+@Setter
```
Comment (karma-o) [inline src/main/java/org/example/bookstore/model/CartItem.java]: let's not forget about equals & hashcode & toString, but exclude lazy initialized fields from them

File: src/main/java/org/example/bookstore/model/CartItem.java [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @JoinColumn(name = "shopping_cart_id", referencedColumnName = "id")
+    @ManyToOne
+    private ShoppingCart shoppingCart;
+
+    @JoinColumn(name = "book_id", referencedColumnName = "id")
+    @ManyToOne
+    private Book book;
```
Comment (karma-o) [inline src/main/java/org/example/bookstore/model/CartItem.java]: let's not leave them with fetch type eager by default and make them lazy

File: src/main/java/org/example/bookstore/service/impl/CartItemServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+    }
+
+    @Override
+    public CartItemDto update(UpdateCartItemRequestDto requestDto, Long cartItemId) {
+        checkIfExistsById(cartItemId);
+        CartItem cartItem = cartItemRepository.findById(cartItemId).get();
+        cartItem.setQuantity(requestDto.quantity());
+        CartItem dbCartItem = cartItemRepository.save(cartItem);
+        return cartItemMapper.toDto(dbCartItem);
+    }
```
Comment (karma-o) [inline src/main/java/org/example/bookstore/service/impl/CartItemServiceImpl.java]: let's also check if that cart item belongs to the user who is logged in to be more secure

**Review by karma-o — CHANGES_REQUESTED** [review]
Looks good! left a few comments to think about

**Review by yevheniisynytsia — APPROVED** [review]

**Pull request merged**